flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDataDeserializationSchema.java

```diff
@@ -206,6 +235,6 @@ public int hashCode() {
                 csvSchema.getArrayElementSeparator(),
                 csvSchema.getQuoteChar(),
                 csvSchema.getEscapeChar(),
-                csvSchema.getNullValue());
+                Arrays.hashCode(csvSchema.getNullValue()));
```

Comment:
why we need to change this ?

Reply:
Because otherwise, we check for the pointer's equality, not equality of the array's content (what we actually want). For a quick test:
```
    public static void main(String[] args) {
        char[] arr1 = { 'x', 'y', 'z' };
        char[] arr2 = { 'x', 'y', 'z' };
        System.out.println("arr1.hashCode(): " + arr1.hashCode() + "; arr2.hashCode(): " + arr2.hashCode());
        System.out.println("Arrays.hashCode(arr1):" + Arrays.hashCode(arr1) + "; Arrays.hashCode(arr1): " + Arrays.hashCode(arr2));
    }
```